hasura.planx.uk/migrations/1683197635599_delete_payment_request_events_on_paid/up.sql

```diff
@@ -0,0 +1,28 @@
+-- function to clean up expiry and reminder emails
+DROP FUNCTION IF EXISTS delete_payment_request_scheduled_email_events;
+
+CREATE OR REPLACE FUNCTION delete_payment_request_scheduled_email_events()
+RETURNS TRIGGER AS $$
+BEGIN
+  DELETE FROM hdb_catalog.hdb_scheduled_events
+  WHERE COMMENT IN (
+    'payment_reminder_' || OLD.id,
+    'payment_reminder_agent_' || OLD.id,
+    'payment_expiry_' || OLD.id,
+    'payment_expiry_agent_' || OLD.id
+  );
+  RETURN NULL;
+END
+$$ LANGUAGE plpgsql VOLATILE;
+
+-- trigger to set up function
+DROP TRIGGER IF EXISTS delete_payment_request_scheduled_email_events_trigger ON payment_requests;
+
+CREATE TRIGGER delete_payment_request_scheduled_email_events_trigger
```

Comment:
nit: could this be `CREATE OR REPLACE TRIGGER...` and therefore negate the need for the `DROP...` command first? Similar question with the function definition!